src/Lexer/DOMLex.hack

```diff
@@ -185,6 +185,25 @@ class HTMLPurifier_Lexer_DOMLex extends HTMLPurifier\HTMLPurifier_Lexer {
 		} else {
 			if ($collect) {
 				$tokens[] = $this->factory->createStart($tag_name, $attr);
+
+				/* If the node is an iframe, we don't want to sanitize the child nodes.
+				* The child nodes should be handled like text.
+				* This block adds the children (and their children and data) to a DOMDocument
+				* in order to convert the children to a string that can be converted into
+				* a HTMLPurifier_Token_Text 
+				*/
+				if ($this->getTagName($node) === 'iframe') {
+					$doc = new \DOMDocument();
+					$children = vec($node->childNodes);
+					foreach ($children as $childNode) {
+						$doc->appendChild($doc->importNode($childNode, true));
+						$node->removeChild($childNode);
+					}
+
+					// Convert the innerHTML to a string and replace trailing "\r\n\r\n" with just "\n"
+					$text = Str\trim_right($doc->saveHTML())."\n";
```

Comment:
> The newline char at least is from here so I wonder if we should sub or check for newline chars before appending it? The other stuff 🤷‍♂️

I'm just bringing this topic here to separate the points. It almost seems like the way I'm converting the nodes to a string is messing with the newline chars... I'm going to see if there's anyway to maintain the same whitespace in this conversion so hopefully we can drop the `."\n"`, but I'll look into some sort of substitution if that's not possible.

Reply:
Alrighty figured it out by checking out the whitespacing on the data v. the whitespacing on the converted text. Turns out `$doc->saveHTML()` was adding a single `\n` indiscriminately, so just slicing the single new `\n` was a better solution instead of chopping off all the trailing whitespace and slapping a new `\n` on.

I was able to take the newline out of the tests, and I also updated the iframe tests with new tests that ensure that the proper whitespacing is maintained.